lib/HLSL/DxilPreparePasses.cpp

```diff
@@ -657,16 +657,23 @@ class DxilFinalizeModule : public ModulePass {
           // In spite of the <=1 assert above, loop here in case the assumption is wrong
           for (auto II = U->user_begin(), EE = U->user_end(); II != EE;) {
             User *UU = *II++;
-            BranchInst *BI = cast<BranchInst>(UU);
-            Function *F = BI->getParent()->getParent();
+            Instruction *I = dyn_cast<Instruction>(UU);
+            if (!I) continue;
```

Comment:
Does this mean the dx.break() intrinsic will stick around and cause a validation error? If so might as well cast<> here and catch the problem early

Reply:
How could it be anything other than an Instruction?  Value to metadata?  Perhaps cast<> is better, as David says.